lte/gateway/python/magma/pipelined/qos/common.py

```diff
@@ -208,67 +210,63 @@ def _setupInternal(self):
-                        del self._qos_store[k]
-
-                    # purge unreferenced qos configs from system
-                    for qos_handle in qos_state:
-                        if qos_handle not in in_store_qid:
-                            LOG.debug("removing qos_handle %d", qos_handle)
-                            self.impl.remove_qos(qos_handle,
-                                qos_state[qos_handle]['direction'], recovery_mode=True)
-
-                    self._initialized = True
-                    LOG.info("init complete with state recovered successfully")
-                except Exception as e:  # pylint: disable=broad-except
-                    # in case of any exception start clean slate
-                    LOG.error("error %s. restarting clean", str(e))
-                    self._clean_restart = True
-                    self.setup()
-
-            asyncio.ensure_future(self.impl.read_all_state(), loop=self._loop).add_done_callback(callback)
+            qos_state = self.impl.read_all_state()
+            try:
+                # populate state from db
+                in_store_qid = set()
+                purge_store_set = set()
+                for k, v in self._redis_store.items():
+                    if v not in qos_state:
+                        purge_store_set.add(k)
+                        continue
+                    in_store_qid.add(v)
+                    _, imsi, ip_addr, rule_num, d = get_key(k)
+                    subscriber = self.get_or_create_subscriber(imsi)
+                    subscriber.update_rule(ip_addr, rule_num, d, v)
+
+                    qid_state = qos_state[v]
+                    if qid_state['ambr_qid'] != 0:
+                        session = subscriber.get_or_create_session(ip_addr)
+                        ambr_qid = qid_state['ambr_qid']
+                        leaf = 0
+                        # its AMBR QoS handle if its config matches with parent
+                        # pylint: disable=too-many-function-args
+                        if self.impl.same_qos_config(d, ambr_qid, v):
+                            leaf = v
+                        session.set_ambr(d, qid_state['ambr_qid'], leaf)
+
+                # purge entries from qos_store
+                for k in purge_store_set:
+                    LOG.debug("purging qos_store entry %s qos_handle", k)
+                    del self._redis_store[k]
+
+                # purge unreferenced qos configs from system
+                for qos_handle in qos_state:
+                    if qos_handle not in in_store_qid:
+                        LOG.debug("removing qos_handle %d", qos_handle)
+                        self.impl.remove_qos(qos_handle,
+                                             qos_state[qos_handle]['direction'], recovery_mode=True)
+
+                LOG.info("init complete with state recovered successfully")
+            except Exception as e:  # pylint: disable=broad-except
```

Comment:
what are the possible errors thrown here? we should aim to be as restrictive as possible while catching exceptions

Reply:
I would like to keep it as as, since this is initialization and we don't was to get into crash loop due to unknown exception.